File: 151051/Maliuk/kr/src/main/java/ru/bsuir/entity/Creator.java
package ru.bsuir.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.bsuir.abstractions.AbstractEntity;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tbl_creator")
public class Creator extends AbstractEntity {

    @Column(nullable = false, unique = true, length = 64)
    @Size(min = 2, max = 64)
    private String login;

    @Column(nullable = false)
    @Size(min = 8, max = 128)
    private String password;

    @Size(min = 2, max = 64)
    private String firstname;

    @Size(min = 2, max = 64)
    private String lastname;

}

File: 251002/Sak/laba2/models/creator.py
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import Text
from .base import Base


class Creator(Base):
    __tablename__ = 'tbl_creator'

    login: Mapped[str] = mapped_column(Text, nullable = False, unique = True)
    password: Mapped[str] = mapped_column(Text, nullable = False)
    firstname: Mapped[str] = mapped_column(Text, nullable = False)
    lastname: Mapped[str] = mapped_column(Text, nullable = False)
   
    # Связь один ко многим с таблицей Story
    stories = relationship("Story", back_populates="creator")